pkg/ipam/crd.go

```diff
@@ -337,14 +337,14 @@ func (n *nodeStore) deleteLocalNodeResource() {
 // on the custom resource passed into the function.
 func (n *nodeStore) updateLocalNodeResource(node *ciliumv2.CiliumNode) {
 	n.mutex.Lock()
-	defer n.mutex.Unlock()
 
 	if n.conf.IPAMMode() == ipamOption.IPAMENI {
 		if err := configureENIDevices(n.ownNode, node, n.mtuConfig); err != nil {
 			log.WithError(err).Errorf("Failed to update routes and rules for ENIs")
 		}
 	}
 
+	var markedToReleaseIPMap = make(map[string]*crdAllocator)
```

Comment:
I would rename the map. 
```suggestion
	var ipsMarkedToRelease = make(map[string]*crdAllocator)
```